### DISCUSSION TITLE
Seeking guidance to adapt the source code for image comparison

### DISCUSSION THREAD

**Discussion by bhupenderbindal** [woelper/oculante#344] [Q&A]
Hello Maintainers,

First of all, thank you. The application is fast when I load a large image, which makes even the native image viewer lag. 

Now, for my case, I plan to use the application to compare images side by side (or up and down) fashion for large-resolution images. The goal is to view, pan and zoom the same section (like a window) of two different images in a synchronous way so one can view the same section of the two images simultaneously, which is difficult as a full image in its original resolution won't even fit on a small screen.

 I am attaching a similar example, where I used smaller images, as the web app consumes memory and gets stuck.
![ezgif-3-3276f82284](https://github.com/woelper/oculante/assets/37266695/810a527e-626a-434e-969f-128b5610f1e6)

I am a beginner in application development, and Rust is also new to me.

Could you guide me with some idea or plan on what changes in the code repo can achieve this?

**Comment by bhupenderbindal** [comment]
Hi,
FYI. 
Recently, I found an [application](https://github.com/olive-groves/butterfly_viewer) that specifically is written for image comparison. This serves the purpose for me now.

**Reply by woelper** [reply]
Hi! Thank you for the great idea! I am happy you found a tool that suits your needs.

I think this is a nice feature request, but there would be substantial work to be done to support multiple images at once - some reasons are editing and the upcoming support for very large images which might require too much memory if we would show multiple ones.

That said, I've been thinking to link several instances of oculante together and sync position, pan and zoom. With that, you could open as many instances you wish and use a central one to control them. Would this be interesting to you?

**Reply by bhupenderbindal** [reply]
Thanks for the reply. It will be a nice feature. However, I think the arrangement of windows/instances will be also required with addition of new instances. The arrangement of images in rows or columns was good in the butterfly-viewer, which I mentioned in the last reply.
